### PR TITLE
fix: properly return http status codes from ingester to querier for RPC function calls

### PR DESCRIPTION
**Pull request by cstyan**
We noticed this earlier with incorrect query syntax on a recording rule that was then sent to an ingester. The error returned to the end user had an HTTP 500 status code without much information because the grpc server was not seeing a valid status code in the result from the ingester, and was wrapping the error as a generic 500 status code.

@slim-bean and I discussed on slack and we think all functions the Querier can call on the Ingester via RPC should use the same error wrapping as we have in our server utils code. Looking at the generated GRPC code, this should be:
- Query
- QuerySample
- Label
- Series
- Tail
- TailersCount
- GetChunkIDs
- GetStats
- GetVolume
- GetDetectedLabels
- GetDetectedFields (this one can never return an error afaict)

Note that I have just done the easy thing for now, and I'll simplify the duplicated code in the morning

### CODE REVIEW

File: pkg/ingester/ingester.go [hunk 1]
```diff
@@ -1111,21 +1120,28 @@ func (i *Ingester) Label(ctx context.Context, req *logproto.LabelRequest) (*logp
 	pprof.Do(ctx, pprof.Labels("path", "read", "type", "labels", "tenant", userID), func(c context.Context) {
 		resp, err = instance.Label(ctx, req, matchers...)
 		if err != nil {
+			fmt.Println("error: ", err)
 			return
 		}
 		if req.Start == nil {
+			fmt.Println("asdf")
```
Comment (cyriltovena) [inline pkg/ingester/ingester.go:1127]:
```suggestion
```

File: pkg/ingester/ingester.go [hunk 2]
```diff
@@ -1111,21 +1120,28 @@ func (i *Ingester) Label(ctx context.Context, req *logproto.LabelRequest) (*logp
 	pprof.Do(ctx, pprof.Labels("path", "read", "type", "labels", "tenant", userID), func(c context.Context) {
 		resp, err = instance.Label(ctx, req, matchers...)
 		if err != nil {
+			fmt.Println("error: ", err)
```
Comment (cyriltovena) [inline pkg/ingester/ingester.go:1123]:
```suggestion
```

File: pkg/ingester/ingester.go [hunk 3]
```diff
@@ -1111,21 +1120,28 @@ func (i *Ingester) Label(ctx context.Context, req *logproto.LabelRequest) (*logp
...
 		if req.Start == nil {
+			fmt.Println("asdf")
+
 			return
 		}
 
 		// Only continue if the active index type is one of async index store types or QueryStore flag is true.
 		asyncStoreMaxLookBack := i.asyncStoreMaxLookBack()
 		if asyncStoreMaxLookBack == 0 && !i.cfg.QueryStore {
+			fmt.Println("qwer")
```
Comment (cyriltovena) [inline pkg/ingester/ingester.go:1135]:
```suggestion
```

File: pkg/ingester/ingester.go [hunk 4]
```diff
@@ -1111,21 +1120,28 @@ func (i *Ingester) Label(ctx context.Context, req *logproto.LabelRequest) (*logp
...
+
 			return
 		}
 
 		var cs storage.Store
 		var ok bool
 		if cs, ok = i.store.(storage.Store); !ok {
+			fmt.Println("1234")
```
Comment (cyriltovena) [inline pkg/ingester/ingester.go:1143]:
```suggestion
```

File: pkg/ingester/ingester.go [hunk 5]
```diff
@@ -1137,18 +1153,24 @@ func (i *Ingester) Label(ctx context.Context, req *logproto.LabelRequest) (*logp
 		start := adjustQueryStartTime(maxLookBackPeriod, *req.Start, time.Now())
 		if start.After(*req.End) {
 			// The request is older than we are allowed to query the store, just return what we have.
+			fmt.Println("ttttt")
```
Comment (cyriltovena) [inline pkg/ingester/ingester.go:1156]:
```suggestion
```

File: pkg/ingester/ingester.go [hunk 6]
```diff
@@ -1137,18 +1153,24 @@ func (i *Ingester) Label(ctx context.Context, req *logproto.LabelRequest) (*logp
...
+
 			return
 		}
 		from, through := model.TimeFromUnixNano(start.UnixNano()), model.TimeFromUnixNano(req.End.UnixNano())
 
 		if req.Values {
 			storeValues, err = cs.LabelValuesForMetricName(ctx, userID, from, through, "logs", req.Name, matchers...)
 			if err != nil {
+				fmt.Println("error2: ", err)
```
Comment (cyriltovena) [inline pkg/ingester/ingester.go:1165]:
```suggestion
```

File: pkg/ingester/ingester.go [hunk 7]
```diff
@@ -1137,18 +1153,24 @@ func (i *Ingester) Label(ctx context.Context, req *logproto.LabelRequest) (*logp
...
+
 				return
 			}
 		} else {
 			storeValues, err = cs.LabelNamesForMetricName(ctx, userID, from, through, "logs", matchers...)
 			if err != nil {
+				fmt.Println("error3: ", err)
```
Comment (cyriltovena) [inline pkg/ingester/ingester.go:1172]:
```suggestion
```

File: pkg/ingester/ingester.go [hunk 8]
```diff
@@ -1160,6 +1182,7 @@ func (i *Ingester) Label(ctx context.Context, req *logproto.LabelRequest) (*logp
 	if err != nil {
 		return nil, err
 	}
+	fmt.Println("resp values: ", resp.Values)
```
Comment (cyriltovena) [inline pkg/ingester/ingester.go:1185]:
```suggestion
```

**Comment by cyriltovena** [comment]
While this is a good idea to have. I think we should investigate how such queries end up on ingester, we should fail fast if this is not good syntax a querier or query-frontend should have caught it.

This was brought up during incident review meeting.

**Comment by cstyan** [comment]
> we should fail fast if this is not good syntax a querier or query-frontend should have caught it.

Agreed, and I will look into this more. Do you have any objections to merging this as is while we continue to investigate why these queries are making it to the ingester in the first place?

**Review by cyriltovena — APPROVED** [review]
LGTM

**Pull request merged**